packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -0,0 +1,745 @@
+
+import { JweEnvelope, JweRecipient } from './JweEnvelope'
+
+@injectable()
+export class AskarWallet implements Wallet {
+  private walletConfig?: WalletConfig
+  private _session?: Session
+
+  private _store?: Store
+
+  private logger: Logger
+  private fileSystem: FileSystem
+
+  private signingKeyProviderRegistry: SigningProviderRegistry
+  private publicDidInfo: DidInfo | undefined
+
+  public constructor(
+    @inject(InjectionSymbols.Logger) logger: Logger,
+    @inject(InjectionSymbols.FileSystem) fileSystem: FileSystem,
+    signingKeyProviderRegistry: SigningProviderRegistry
+  ) {
+    this.logger = logger
+    this.fileSystem = fileSystem
+    this.signingKeyProviderRegistry = signingKeyProviderRegistry
+  }
+
+  public get isProvisioned() {
+    return this.walletConfig !== undefined
+  }
+
+  public get isInitialized() {
+    return this._store !== undefined
+  }
+
+  public get publicDid() {
+    return this.publicDidInfo
+  }
+
+  public get store() {
+    if (!this._store) {
+      throw new AriesFrameworkError(
+        'Wallet has not been initialized yet. Make sure to await agent.initialize() before using the agent.'
+      )
+    }
+
+    return this._store
+  }
+
+  public get session() {
+    if (!this._session) {
+      throw new AriesFrameworkError('No Wallet Session is opened')
+    }
+
+    return this._session
+  }
+
+  public get masterSecretId() {
+    if (!this.isInitialized || !(this.walletConfig?.id || this.walletConfig?.masterSecretId)) {
+      throw new AriesFrameworkError(
+        'Wallet has not been initialized yet. Make sure to await agent.initialize() before using the agent.'
+      )
+    }
+
+    return this.walletConfig?.masterSecretId ?? this.walletConfig.id
+  }
+
+  /**
+   * Dispose method is called when an agent context is disposed.
+   */
+  public async dispose() {
+    if (this.isInitialized) {
+      await this.close()
+    }
+  }
+
+  /**
+   * @throws {WalletDuplicateError} if the wallet already exists
+   * @throws {WalletError} if another error occurs
+   */
+  public async create(walletConfig: WalletConfig): Promise<void> {
+    await this.createAndOpen(walletConfig)
+    await this.close()
+  }
+
+  /**
+   * @throws {WalletDuplicateError} if the wallet already exists
+   * @throws {WalletError} if another error occurs
+   */
+  public async createAndOpen(walletConfig: WalletConfig): Promise<void> {
+    this.logger.debug(`Creating wallet '${walletConfig.id}`)
+
+    const askarWalletConfig = await this.getAskarWalletConfig(walletConfig)
+    try {
+      this._store = await Store.provision({
+        recreate: false,
+        uri: askarWalletConfig.uri,
+        profile: askarWalletConfig.profile,
+        keyMethod: askarWalletConfig.keyMethod,
+        passKey: askarWalletConfig.passKey,
+      })
+      this.walletConfig = walletConfig
+      this._session = await this._store.openSession()
+
+      // TODO: Master Secret creation (now part of IndyCredx/AnonCreds)
```

Comment:
Yeah I want to extract this from the wallet as it is more an AnonCreds thing. See #1198